src/paginationCalls/pageDescriptor.ts

```diff
@@ -103,6 +116,96 @@ export class PageDescriptor implements Descriptor {
     return stream;
   }
 
+  /**
+   * Create an async iterable which can be recursively called for data on-demand.
+   */
+  asyncIterate(
+    apiCall: GaxCall,
+    request: RequestType,
+    options: CallSettings
+  ): AsyncIterable<{} | undefined> {
+    const iterable = this.createIterator(options);
+    const funcPromise =
+      typeof apiCall === 'function' ? Promise.resolve(apiCall) : apiCall;
+    funcPromise
+      .then((func: GaxCall) => {
+        this.makeCall(request, func, options);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+    return iterable;
+  }
+
+  createIterator(options: CallSettings): AsyncIterable<{} | undefined> {
+    const self = this;
+    const cache = this.cache;
```

Comment:
nit: remove this constant, just use `self.cache` everywhere below. Will make the code a little bit more readable.